dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/SPIMetaData.java

```diff
@@ -27,24 +27,34 @@
 import io.openliberty.microprofile.telemetry.internal.rest.TelemetryClientFilter;
 import io.openliberty.microprofile.telemetry.internal.rest.TelemetryContainerFilter;
 
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
 @Component(service = CDIExtensionMetadata.class, configurationPolicy = IGNORE)
 public class SPIMetaData implements CDIExtensionMetadata, CDIExtensionMetadataInternal {
 
+    private static final TraceComponent tc = Tr.register(SpanProxy.class);
+
     @Override
     public Set<Class<?>> getBeanClasses() {
-        Set<Class<?>> beans = new HashSet<Class<?>>();
-        beans.add(OpenTelemetryProducer.class);
-        if (!AgentDetection.isAgentActive()) {
-            beans.add(WithSpanInterceptor.class);
+        try {
+            Set<Class<?>> beans = new HashSet<Class<?>>();
+            beans.add(OpenTelemetryProducer.class);
+            if (!AgentDetection.isAgentActive()) {
+                beans.add(WithSpanInterceptor.class);
+            }
+            beans.add(TelemetryClientFilter.class);
+            beans.add(TelemetryContainerFilter.class);
+            return beans;
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return new HashSet();
```

Comment:
I don't think it's necessary to add it for this class. We always return the same things and this error would occur at startup.